packages/client/hmi-client/src/components/workflow/tera-model-node.vue

```diff
@@ -48,15 +53,18 @@ const model = ref<Model | null>();
 const initialValues = ref<StringValueMap>({});
 const parameterValues = ref<StringValueMap>({});
 
+const thisModelOperation = cloneDeep(modelOperation);
+
 function run() {
-	if (ModelOperation.action) {
+	if (thisModelOperation.action) {
 		console.log(
-			ModelOperation.action({
+			thisModelOperation.action({
 				model: model.value,
 				initialValues: initialValues.value,
 				parameterValues: parameterValues.value
 			})
 		);
+		emit('append-output-port', props.node.id, thisModelOperation.outputs[0].type);
```

Comment:
I think we should refactor our node components, I don't think `tera-model-node` should have to emit an event to add a port, nor should the node object be passed as a prop into *both* `tera-workflow-node` and `tera-model-node`

Reply:
Two separate observations.

@echl: 
Because we generally treat props as a one-way data-bind, we don't typically want to directly change `props.workflowNode.xyz`. So what we have here follows the data-down-action-up paradigm ... which is fine in my opinion. In leu of the emit, we can consider doing a v-model paradigm that is closer to a two-way-data-bind. Do you have examples you want to share in terms of how you want to do things or are we okay with this for now?


Let's talk about the tera-model-node  vs era-workflow-node issue offline, it is largely not relevant to this PR.

Reply:
I'm okay with it for now. Something about it seems unintuitive to me but I'll explore it further.